### PR TITLE
Improve definition of text markup escape sequences

### PR DESCRIPTION
**Pull request by henrikt-ma**
This is the first of two smaller issues detected in MCP-0031, which is currently in the stage _Under evaluation_.  It is our belief that the MCP is still is sufficiently good shape to proceed with the voting for its acceptance, but that — if the MCP is accepted — we should try to address these issues before the release of the next version of the Modelica Specification.

This PR sorts out some details regarding the use of the (percent-encoded) text markup escape sequences.  Most notably:
- It becomes possible to express a closing bracked (`]`) inside a link text.
- Text markup escape sequences are disabled in some contexts where they are not needed and where not having them makes manual editing easier.

### CODE REVIEW

File: chapters/annotations.tex [hunk 1]
```diff
@@ -227,35 +248,49 @@ \subsubsection{Variable Replacements}\label{variable-replacements}
...
+Links take the form \lstinline!%[$\mathit{text}$]($\mathit{link}$)!, where the \lstinline![$\mathit{text}$]! part is optional.  Text markup escape sequences don't apply inside the $\mathit{link}$, which can be in either of the following forms:
 \begin{itemize}
-\item A URI, such as
-\lstinline!https://github.com/modelica/ModelicaSpecification! or
-\lstinline!modelica://Modelica.Blocks!.
-\item A \lstinline!variable:<id>!, where \lstinline!<id>! is a component
-reference such as \lstinline!inertia1.w!.
-\item A \lstinline!plot:<id>!, where \lstinline!<id>! is the identifier of a
-\lstinline!Plot! in the current \lstinline!Figure!.
+% Use very short Modelica URI combined with carefull chosen wording below to deal with difficult line breaking.
```
Comment (maltelenz) [inline chapters/annotations.tex:275]: Typo in `carefully`

**Review by maltelenz — APPROVED** [review]
Looks good to me.

File: chapters/annotations.tex (hunk 1 above)
Comment (maltelenz) [inline chapters/annotations.tex:275]:
```suggestion
% Use very short Modelica URI combined with carefully chosen wording below to deal with difficult line breaking.
```

File: chapters/annotations.tex [hunk 2]
```diff
@@ -204,14 +205,34 @@ \subsubsection{Plot Curves}\label{plot-curves}
...
-In most places where text for display is defined the final value of a result
-variable can be embedded by refering to the variable as
-\lstinline!%{inertia1.w}!.  This is similar to the \lstinline!Text! graphical primitive
-in \cref{text}.
+
+In the places listed in \cref{attributes-with-variable-replacements} where text for display is defined, the final value of a result variable can be embedded by refering to the variable as \lstinline!%{inertia1.w}!.  This is similar to the \lstinline!Text! graphical primitive in \cref{text}.
 
 \begin{table}[H]
 \caption{Attributes that can use variable replacements.}
+\label{attributes-with-variable-replacements}
```
Comment (HansOlsson) [inline chapters/annotations.tex:235]: I normally try to use the exact text for sections. That would be attributes-that-can-use-variable-replacements not attributes-with-variable-replacements

File: chapters/annotations.tex [hunk 3]
```diff
@@ -227,35 +248,49 @@ \subsubsection{Variable Replacements}\label{variable-replacements}
...
 parameters and constants, so that values to be shown in a caption can be
 determined during simulation.
 
-The percent character is encoded \lstinline!%%!.  Neither \lstinline!%class! nor
-\lstinline!%name! is supported in this context, as this information is expected
-to already be easily accessible (when applicable) in tool-specific ways. (Titles
-making use of \lstinline!%class! or \lstinline!%name! would then only lead to
-ugly duplication of this information.)
+\begin{nonnormative}
+By design, neither \lstinline!%class! nor \lstinline!%name! is supported in this context, as this information is expected to already be easily accessible (when applicable) in tool-specific ways.  (Titles making use of \lstinline!%class! or \lstinline!%name! would then only lead to ugly duplication of this information.)
```
Comment (HansOlsson) [inline chapters/annotations.tex:266]: I find ugly a bit too strong and would prefer unnecessary, or some formulation with redundant.

File: chapters/annotations.tex [hunk 4]
```diff
@@ -227,35 +248,49 @@ \subsubsection{Variable Replacements}\label{variable-replacements}
...
+
+\begin{nonnormative}
+Note that for the character `]' to appear in $\mathit{text}$, it needs to be encoded as the escape sequence \lstinline!%]!, or it would be interpreted as the terminating delimiter of the \lstinline![$\mathit{text}$]!.
+
+Similarly, the closing parenthesis `)' must be handled with care in $\mathit{link}$ in order to not be interpreted as the terminating delimiter of the \lstinline!($\mathit{link}$)!.
+\begin{itemize}
+\item For a URI, a closing parenthesis must be URL encoded in order to not be interpreted as the end of the \lstinline!($\mathit{link}$)!.  For example, the URL in \lstinline[language={[nocomment]modelica}]!%(http://bad.org/(tryme))! is just \filename{http://bad.org/(tryme}, and the entire link is followed by a stray closing parenthesis.  To make it work, one has to use URL encoding: \lstinline[language={[nocomment]modelica}]!%(http://bad.org/%28tryme%29)! (using URL encoding of the opening parenthesis just for symmetry, and note that the \lstinline!%! of the percent-encoded sequences are not subject to text markup escape sequences).
```
Comment (HansOlsson) [inline chapters/annotations.tex:288]: Using something that possibly could be a valid url as example is error-prone. Shouldn't we follow RFC 2606 and replace bad.org by example.org?

File: chapters/annotations.tex (hunk 2 above)
Comment (henrikt-ma) [reply chapters/annotations.tex:235]: But this isn't even a title, it's a caption, and a caption could even consist of several paragraphs.  I mean, that I like short and stable labels probably doesn't come as a surprise, but while the scheme of systematic mapping of section titles to labels could sort of work (especially if one decides to never change a section title), systematic mapping of captions… really?

File: chapters/annotations.tex (hunk 3 above)
Comment (henrikt-ma) [reply chapters/annotations.tex:266]: Can one say _redundant duplication_, or is _redundant_ redundant in that expression?
> Titles making use of \lstinline!%class! or \lstinline!%name! would then result in redundant duplication of this information.

File: chapters/annotations.tex (hunk 4 above)
Comment (henrikt-ma) [reply chapters/annotations.tex]: Sure.  Fixed.  Resolving.

File: chapters/annotations.tex (hunk 3 above)
Comment (HansOlsson) [reply chapters/annotations.tex:266]: But 'result in redundant information' would work, right?

Comment (henrikt-ma) [reply chapters/annotations.tex:266]: It removes the local redundancy around the word _redundant_, but lacks the explanatory effect of the word _duplication_ with its negative connotation.  Without the explanatory effect, I find the entire sentence redundant in view of the previous sentence.

**Pull request merged**